### ISSUE TITLE
The extension doesn't find the `lldb-dap.exe` DAP in the toolchain (its new name)

### ISSUE THREAD

**Issue by tristanlabelle** [swift-server/vscode-swift#737]
**Describe the bug**
The debug adapter is not found in more recent toolchains because the DAP executable has been renamed.

https://github.com/swift-server/vscode-swift/blob/05aec075052dc007164bd6b012c1ab2931a9a324/src/debugger/debugAdapter.ts#L47

**To Reproduce**
1. Install a main branch Swift toolchain from the past few months
2. Set `"swift.debugger.useDebugAdapterFromToolchain": true,`
3. Unset `"swift.debugger.path"`
3. Start debugging a test
4. Notice that codelldb is silently used as the debugger

**Expected behavior**
The extension should support both names, favoring `lldb-dap.exe` when present.

**Environment**
 - OS: Windows 11
 - Swift version:
 ```
compnerd.org Swift version 5.11-dev (LLVM 1bcf1d71b715d0d, Swift a1d70b6086b16bb)
Target: x86_64-unknown-windows-msvc
 ```
 - Visual Studio Code version: 1.88.0
 - vscode-swift version: 1.8.1

**Comment by adam-fowler** [comment]
As I remember @compnerd was going to fix this but I can do this. What are your thoughts on using lldb-dap.exe for swift version >= 6? I know that will mean earlier Swift 6 nightlies will fail, but I don't really want to have to base the decision on whether swift-dap.exe or swift-vscode.exe exists.

**Comment by tristanlabelle** [comment]
My take is that testing for features is generally preferable to testing for version numbers because it makes the dependency more obvious. We can also pick this up, let me know. (fyi @z2oh)

**Comment by adam-fowler** [comment]
This is now released in v1.9
